File: android/app/src/main/java/com/reviewanything/app/data/model/Chunk.kt
package com.reviewanything.app.data.model

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "chunks",
    foreignKeys = [
        ForeignKey(
            entity = Note::class,
            parentColumns = ["id"],
            childColumns = ["noteId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class Chunk(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val noteId: Int = 0,
    val content: String = "",
    val contentHash: String = "",
    val headingPath: String = "",
    val createdAt: Long = System.currentTimeMillis()
)

File: windows/ReviewAnything/Models/Chunk.cs
namespace ReviewAnything.Models;

public class Chunk
{
    public int Id { get; set; }
    public int NoteId { get; set; }
    public Note Note { get; set; } = null!;
    public string Content { get; set; } = "";
    public string ContentHash { get; set; } = "";
    public string HeadingPath { get; set; } = "";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public List<ReviewItem> ReviewItems { get; set; } = new();
}
